workflow/models.py

```diff
@@ -1,63 +1,198 @@
+# -*- coding: utf-8 -*-
+"""Workflow Models"""
+
 from __future__ import unicode_literals
 from django.db import models
 from django.contrib.auth.models import AbstractUser
 from django.contrib.postgres.fields import JSONField
 
 
 class Workflow(models.Model):
+    """Workflow Model"""
+
     id = models.AutoField(primary_key=True)
     title = models.CharField(max_length=128, blank=False)
     description = models.TextField(null=False, blank=True)
     initialization_values = JSONField()
 
+    class JSONAPIMeta:
+        resource_name = 'workflows'
+
     def __str__(self):
         return self.title
 
+
 class Section(models.Model):
+    """Section Model"""
+
     id = models.AutoField(primary_key=True)
     label = models.CharField(max_length=128, blank=False, null=False)
     description = models.TextField(null=False, blank=True)
-    workflow = models.ForeignKey('Workflow', related_name='sections', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='sections',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'sections'
+
     def __str__(self):
         return self.label
 
+
 class Widget(models.Model):
+    """Widget Model"""
+
     id = models.AutoField(primary_key=True)
     label = models.CharField(max_length=128, null=False, blank=False)
     description = models.TextField(null=False, blank=True)
     widget_type = models.CharField(max_length=24, blank=False, null=False)
-    parameter_mappings = models.ManyToManyField('WidgetParameterMapping', related_name='consumer_widgets')
-    section = models.ForeignKey('Section', related_name='widgets', null=False)
-    workflow = models.ForeignKey('Workflow', related_name='widgets', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    section = models.ForeignKey(
+        'Section',
+        related_name='widgets',
+        null=False
+    )
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='widgets',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'widgets'
+
     def __str__(self):
         return self.label
 
-class WidgetParameterMapping(models.Model):
+
+class ParameterAlias(models.Model):
```

Comment:
Parameter alias lets a widget refer to a parameter by different names, so that internal to the widget, we don't have to guess or do wacky lookups to find the place to store the data a user is entering.